taf/tests/test_updater/update_utils.py

```diff
@@ -256,3 +257,112 @@ def _update_expect_error():
+    )
+
+    try:
+        check_if_commits_match(client_repos, origin_auth_repo.path.parent.parent)
+    except AssertionError:
+        return False
+    assert True, "The repositories are unexpectedly in sync."
+
+
+def verify_client_repos_state(client_dir: Path, auth_repo: AuthenticationRepository):
+    """
+    Verify that the client's repositories are in the correct state.
+    This means that the target repositories in the client repo should be in sync with the origin repo,
+    and the client's auth repo should be updated to the last validated commit.
+    """
+    client_auth_repo = AuthenticationRepository(path=client_dir / auth_repo.name)
+    client_target_repos = load_target_repositories(auth_repo, library_dir=client_dir)
+
+    check_last_validated_commit(client_auth_repo.path)
+
+    successful_update = True
+    for repo_name, client_repo in client_target_repos.items():
+        origin_repo = GitRepository(auth_repo.path.parent.parent, repo_name)
+        client_commit = client_repo.head_commit_sha()
+        origin_commit = origin_repo.head_commit_sha()
+
+        if client_commit != origin_commit:
+            successful_update = False
+            break
+
+    if successful_update:
+        check_if_commits_match(client_target_repos, auth_repo.path.parent.parent)
+    else:
+        for repo_name, client_repo in client_target_repos.items():
+            origin_repo = GitRepository(auth_repo.path.parent.parent, repo_name)
+            client_commit = client_repo.head_commit_sha()
+
+            # Get the target file path in the client repo
+            target_file_path = client_repo.path / "test1.txt"
+
+            # Ensure the target file exists
+            if not target_file_path.is_file():
+                raise FileNotFoundError(
+                    f"Target file {target_file_path} does not exist"
+                )
+
+            # Extract commit SHA from the target file in the client repo
+            target_commit_sha = client_repo._git(
+                f'log -n 1 --pretty=format:"%H" -- {target_file_path}'
+            ).strip('"')
+
+            # Assert that the top commits of target repositories are the same as the commit SHA specified in the corresponding target files
+            assert (
+                client_commit == target_commit_sha
+            ), f"Target repo {repo_name} should have the same top commit as specified in the corresponding target file"
+            assert (
+                client_commit != origin_commit
+            ), f"Target repo {repo_name} should not have the same commit as specified in the client's auth repo"
+
+
+def verify_partial_update(
+    client_dir: Path, origin_auth_repo: AuthenticationRepository, original_commits: dict
+):
+    """
+    Verify that the client's repositories are in the correct state following a partial update.
+    This means that the top commits of target repositories are the same as the top commits of origin repositories,
```

Comment:
The opposite should be true. Top commits of the client's local target repositories, in case of a partial update, should be different from the top commits of the origin repos (partial update is an unsuccessful update, which failed because the origin repos contained invalid commits). If a user manually pulled invalid commits, these tests need to verify that the the updater is able to remove them. Repositories should be at the most recent valid commit